### PR TITLE
llvm: add new versions and set default for libomptarget according to os

### PR DESCRIPTION
**Pull request by s-sajid-ali**
On a mac with `+libomptarget`, one gets :
```
libomptarget cannot be built on Windows and MacOS X!
```
on os x.

Reference: https://github.com/llvm/llvm-project/blob/48bb7bb868919a090492bffb9f3c170dae5250eb/openmp/CMakeLists.txt#L101-L112

Additionally, add checksums for new versions.

### CODE REVIEW

**Comment by naromero77** [comment]
New versions are OK, but there is a recent merge that created conflicts that now need to be resolved.

**Comment by s-sajid-ali** [comment]
Rebased over develop.

**Comment by naromero77** [comment]
@skosukhin Could you please review?

File: var/spack/repos/builtin/packages/llvm/package.py
Comment (skosukhin) [inline var/spack/repos/builtin/packages/llvm/package.py]: I'd rather introduce conflicts:
```suggestion
    variant("libomptarget", default=True, description="Build the OpenMP offloading library")
    conflicts("+libomptarget", when="~clang")
    for _p in ["darwin", "windows"]:
        conflicts("+libomptarget", when="platform={0}".format(_p))
    del _p
```

**Review by naromero77 — APPROVED** [review]
Looks good to me.

Will wait for additional comments/approval from @skosukhin.

**Review by skosukhin — APPROVED** [review]

**Pull request merged**
